test/model/package.js

```diff
@@ -0,0 +1,75 @@
+////////////////////////////////////////////////////////////////////////////////
+// TEST UTILITIES
+// Primarily, this file contains a test utility to make testing chained knex
+// expressions easier. But it may grow to include more database testing
+// utilities eventually.
+
+const should = require('should');
+
+// Create a standard knex object, then replace all methods we find with a test
+// harness.
+const knex = require('knex')({ client: 'pg' });
+const shim = (methodName) => function(...args) {
+  this.__calls.push([ methodName, args ])
+  return this;
+};
+for (const method in knex)
+  if (typeof knex[method] === 'function')
+    knex[method] = shim(method);
+
+// Also, insert Promise methods.
+for (const method of [ 'then' ])
```

Comment:
Wow, `for... of`... that's fun!

Reply:
It is technically still [20% slower](https://jsperf.com/cxlt-iterations) than a pure `for(;;)` loop buuuuut eh? They're tests.